sbtc/src/testing/regtest.rs

```diff
@@ -99,6 +99,36 @@ pub fn initialize_blockchain() -> (&'static Client, &'static Faucet) {
     (rpc, faucet)
 }
 
+/// See `initialize_blockchain`
+pub fn initialize_blockchain_devenv() -> (&'static Client, &'static Faucet) {
+    static BTC_CLIENT: OnceLock<Client> = OnceLock::new();
+    static FAUCET: OnceLock<Faucet> = OnceLock::new();
+    let rpc = BTC_CLIENT.get_or_init(|| {
+        let username = BITCOIN_CORE_RPC_USERNAME.to_string();
+        let password = BITCOIN_CORE_RPC_PASSWORD.to_string();
+        let auth = Auth::UserPass(username, password);
+        Client::new(
+            &format!("http://127.0.0.1:18443/wallet/{BITCOIN_CORE_WALLET_NAME}"),
+            auth,
+        )
+        .unwrap()
+    });
+
+    let faucet = FAUCET.get_or_init(|| {
+        get_or_create_wallet(rpc, BITCOIN_CORE_WALLET_NAME);
+        let faucet = Faucet::new(FAUCET_SECRET_KEY, AddressType::P2wpkh, rpc);
+        faucet.track_address(FAUCET_LABEL);
+
+        // We cannot create 100 blocks here, as it may break the Stacks signers.
+        // So we just assume someone already funded the faucet a bit to survive
```

Comment:
I wonder if you could instead add a wait-loop here that checks for 100 blocks with an optional timeout as a function arg? Just thinking it might help potential users of the function and/or help ensure consistency in tests.

Reply:
The issue is that the test requires to stop the bitcoin miner to avoid interferences, so based on when it was stopped we may never reach enough blocks so we would still need some "outside" intervention. That sucks, but expecting some funding happening beforehand as test setup seemed the less painful approach.

Reply:
Ah okay, could we make the "limitations" clear on the function docs so that it's visible externally to this file?

Reply:
Oh right, makes sense, added in [118bedf](https://github.com/stacks-sbtc/sbtc/pull/1585/commits/118bedfaeb5a80bee969e750a721639e6f7c63f6)